Expose all build suggestion APIs in `index.ts`

james0888215 opened the pull request:
<!---
Thanks for helping to make Cucumber better! 💖

You can feel free to open a "Draft" status pull request if you're not ready for feedback yet.

Don't worry about getting everything perfect! We're here to help and will coach you through
to getting your pull request ready to merge.

The prompts below are for guidance to help you describe your change in a way that is most 
likely to make sense to other people when they are reviewing it. Still, it's just a guide, 
so feel free to delete anything that doesn't feel appropriate, and add anything additional 
that seems like it would provide useful context. 👏🏻
-->

### 🤔 What's changed?

<!-- Describe your changes in detail -->
Added buildSuggestionFromCucumberExpression and buildSuggestionsFromRegularExpression to suggestions index in order to expose them directly from the package import.

### ⚡️ What's your motivation? 

Allows access to the buildSuggestionFromCucumberExpression  function directly.
<!-- 
What motivated you to propose this change? Does it fix a bug? Add a new feature?
If it fixes an open issue, you can link to the issue here, e.g. "Fixes #99"
-->

### 🏷️ What kind of change is this?

<!--- Delete any options that are not relevant -->

- :bank: Refactoring/debt/DX (improvement to code design, tooling, etc. without changing behaviour)

### ♻️ Anything particular you want feedback on?

<!-- 
Is there anything in this change you're unsure about, or would 
particularly like reviewers to give you feedback on?
-->

### 📋 Checklist:

<!--- 
This is to help you remember all the little things we often forget to do!

Feel free to delete any tasks that are not relevant, or add new ones.
-->

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

kieran-ryan reviewed (changes requested):
Cheers @james0888215! Just requires an [`npm run eslint-fix`](https://github.com/cucumber/language-service/blob/a6401ebf61fdf5621145493b936fc548c966f0b7/package.json#L38) (due to alphabetic ordering of singular `buildSuggestionFromCucumberExpression`) and a changelog entry [under the `Added` section](https://github.com/cucumber/language-service/blob/a6401ebf61fdf5621145493b936fc548c966f0b7/CHANGELOG.md?plain=1#L11).

You might advise whether is possible to meet your particular use case at present through calling `buildSuggestions` (which should ideally act as a factory). Just thinking whether possible through existing API or if any present gap or use case to address - including whether any suggested changes for that API?

james0888215 commented:
Thanks for the feedback @kieran-ryan. Updated imports with es-lint and updated changelog

Regarding the use case, for my own case it provides better granular control on how to actually build out the suggestion based on the type of expression I know I am working with. It also just allows users of the service to be able to access these functions without directly referencing in the node_modules.

Would be happy to raise a separate PR for use case section around both functions also.

kieran-ryan reviewed (approved):
Cheers @james0888215 - good to go! 🚀
